internal/pkg/gateway/api.go

```diff
@@ -321,7 +337,7 @@ func (gs *Server) Endorse(ctx context.Context, request *gp.EndorseRequest) (*gp.
 		return nil, newRpcError(codes.Aborted, "failed to collect enough transaction endorsements, see attached details for more info", errorDetails...)
 	}
 
-	env, err := protoutil.CreateTx(proposal, endorsements...)
+	env, err := protoutil.CreateTx(header, proposal.Payload, logger.With("channel", channel, "txID", request.TransactionId), endorsements...)
```

Comment:
Above you changed the function name to CreateSignedTxWithLogger() when you added the logger. Here you add the logger without changing the function name. Better to be consistent.